jigglypuffRL/common/trainer.py

```diff
@@ -0,0 +1,210 @@
+import os
+
+import gym
+import torch
+import numpy as np
+
+from jigglypuffRL.common import (
+    set_seeds,
+    Logger
+)
+from abc import ABC
+
+class Trainer(ABC):
+    def __init__(self, agent, env, logger, buffer=None, off_policy=False, save_interval=0,
+                 render=False, max_ep_len=1000, distributed=False, ckpt_log_name='experiment',
+                 steps_per_epoch=4000, epochs=10, device='cpu', log_interval=10, batch_size=50,
+                 seed=None, deterministic_actions=False):
```

Comment:
Does Trainer require all these parameters? Isn't it enough if you just pass these to the inherited classes?

Reply:
Yup, you're right, I'll change that.